projects/igniteui-angular/src/lib/services/transaction/igx-hierarchical-transaction.ts

```diff
@@ -51,13 +51,24 @@ export class IgxHierarchicalTransactionService<T extends HierarchicalTransaction
         }
     }
 
-    public commit(data: any[], primaryKey?: any, childDataKey?: any): void {
-        if (childDataKey) {
-            DataUtil.mergeHierarchicalTransactions(data, this.getAggregatedChanges(true), childDataKey, primaryKey, true);
+    /**
+     * Applies all transactions over the provided data
+     * @param data Data source to update
+     * @param primaryKey Primary key of the hierarchical data
+     * @param childDataKey Kye of child data collection
+     * @param id Optional record id to commit transactions for
+     */
+    public commit(data: any[], primaryKey?: any, childDataKey?: any, id?: any): void {
```

Comment:
I realize this is not a change from this PR, but.. Not a great fan of the mismatched signature here 😐 
Might be worth opening a new issue to look into that, cuz this service is still marked as hidden/experiamental but the feature is live and the HGrid doesn't override the transactions property, so you get auto-complete for the base type.

Reply:
this is for new issue 😏